### ISSUE TITLE
Doctrine class metadata not found for extended interface

### ISSUE THREAD

**Issue by teohhanhui** [Sylius/Sylius#10408]
**Sylius version affected**: 1.x

**Description**  
For any resource, it should be possible to get the Doctrine class metadata using any of the interfaces for the resource defined at the component / core / app level.

**Steps to reproduce**  
1. Create a new project from Sylius-Standard.
2. Try to get Doctrine class metadata using `Sylius\Component\Core\Model\ProductInterface`.

**Possible Solution**  
I think there is a design problem.

If you look at https://github.com/Sylius/SyliusResourceBundle/blob/v1.5.0/src/Bundle/DependencyInjection/Compiler/DoctrineTargetEntitiesResolverPass.php

it does not go up the inheritance chain - and it's not safe to do so, as we can't tell which interfaces should be mapped, and which should not. So if we set `interface` to a child interface, things would stop working if you try using the parent interface(s).

What is actually expected is that all of the interfaces should work, e.g.:

- `Sylius\Component\Product\Model\ProductInterface`
- `Sylius\Component\Core\Model\ProductInterface`
- `App\Model\ProductInterface` (for example)

So we need a way to map multiple interfaces.

**Comment by vvasiloi** [comment]
I think going up or down the inheritance chain isn't a good idea.
What if an interface is extended/implemented by multiple interfaces/classes?
Like `Sylius\Component\Core\Model\ProductInterface` is implemented by `App\Entity\Product`
and also extended by `App\Model\FooProductInterface`, which is then implemented by `App\Entity\FooProduct`.
You will end up having either `App\Model\FooProductInterface` mapped to `App\Entity\Product` or `Sylius\Component\Core\Model\ProductInterface` mapped to `App\Entity\FooProduct`, depending on the order they're processed.

Maybe it could be done by having a way to explicitely define multiple/additional interfaces for a class.
WDYT?

**Comment by teohhanhui** [comment]
Yes, that's what I've mentioned. We cannot go up / down the inheritance chain as that's unsafe. So we must provide a way to map multiple interfaces. :smile:

**Comment by vvasiloi** [comment]
Oh, sorry, I read that poorly. 😄 
So, maybe something like that:
```
// Sylius\Bundle\ResourceBundle\DependencyInjection\Configuration::addResourcesSection
->arrayNode('interfaces')
    ->cannotBeEmpty()
    ->scalarPrototype()
    ->beforeNormalization()->castToArray()
 ->end()
```
and 
```
// Sylius\Bundle\ResourceBundle\DependencyInjection\Compiler\DoctrineTargetEntitiesResolverPass::getInterfacesMapping

private function getInterfacesMapping(array $resources): array
{
    $interfaces = [];
    foreach ($resources as $alias => $configuration) {
        // current code
        if (isset($configuration['classes']['interfaces'])) {
            foreach ($configuration['classes']['interfaces'] as $interface) {
                $interfaces[$interface] = sprintf('%s.model.%s.class', $alias[0], $alias[1]);
            }
        }
    }

    return $interfaces;
}
```
Looks too easy to be true though.
